File: tiq/math/fizzbuzz.go
package math

import (
	"fmt"
	"strconv"
)

// Write a program that outputs the string representation of numbers from 1 to
// n. But for multiples of three it should output “Fizz” instead of the number
// and for the multiples of five output “Buzz”. For numbers which are multiples
// of both three and five output “FizzBuzz”.
func fizzBuzz(n int) []string {
	res := make([]string, n)
	for i := 0; i < n; i++ {
		if x := i + 1; x%15 == 0 {
			res[i] = "FizzBuzz"
		} else if x%5 == 0 {
			res[i] = "Buzz"
		} else if x%3 == 0 {
			res[i] = "Fizz"
		} else {
			res[i] = strconv.Itoa(x)
		}
	}
	return res
}

// Run Fizzbuzz
func RunFizzBuzz() {
	ans := fizzBuzz(100)
	fmt.Println(ans)
}

File: tiq/math/fizzbuzz.cpp

#include <vector>
#include <string>
#include <iostream>
#include <numeric>

using namespace std;

class Solution
{
public:
    void operator()()
    {
        auto v = fizzbuzz(20);
        for (auto const& x : v) {
            cout << x << ' ';
        }
        cout << '\n';
    }

    vector<string> fizzbuzz(int n)
    {
        vector<string> result(n);
        for (int i = 0; i < n; i++)
        {
            auto x = i + 1;
            if (x % 15 == 0)
            {
                result[i] = "FizzBuzz";
            }
            else if (x % 3 == 0)
            {
                result[i] = "Fizz";
            }
            else if (x % 5 == 0)
            {
                result[i] = "Buzz";
            }
            else
            {
                result[i] = to_string(x);
            }
        }
        return result;
    }
};